src/controllers/periods.ts

```diff
@@ -1,18 +1,63 @@
-import { Request, Response } from 'express';
 import PeriodModel from '@entities/Period';
-import { QueryInput } from '@shared/inputs';
 import { getQuerySort } from '@shared/functions';
+import { QueryInput } from '@shared/inputs';
+import { Request, Response } from 'express';
+import { StatusCodes } from 'http-status-codes';
 
-// get periods
-const getPeriods = async (
+export const all = async (
   req: Request<any, QueryInput, any>,
   res: Response
 ): Promise<Response> => {
   const response = await PeriodModel.paginate({
     ...req.query,
     sort: getQuerySort(req.query),
   });
-  return res.status(200).json(response);
+  return res.status(StatusCodes.OK).json(response);
 };
 
-export default { getPeriods };
+export const single = async (
+  req: Request<any, QueryInput, any>,
+  res: Response
+): Promise<Response> => {
+  let period = await PeriodModel.findById(req.params.periodId);
+  if (!period) return res.status(StatusCodes.INTERNAL_SERVER_ERROR);
+  return res.status(StatusCodes.OK).json(period);
+};
+
+export const update = async (
+  req: Request<any, QueryInput, any>,
+  res: Response
+): Promise<Response> => {
+  const name = req.query.name as string;
+  const endDate = req.query.endDate as string;
+
+  if (!name && !endDate) return res.status(StatusCodes.BAD_REQUEST);
```

Comment:
i created `QueryInput` for the query params that will be used for list endpoint, for exaple limit, page, etc :
```export interface QueryInput extends Query {
  sortColumn?: string;
  sortType?: string;
  limit?: string;
  page?: string;
}```

i though that on update params will go to the body of the request. if so we can create periodUpdateParams interface inside `inputs.ts` and than request could look like this:
```Request<RequestBody, any, any>```

That way we know exactly how request will look like and we don't need to check if `name` and `endDate` exist inside the request

Reply:
Ok, all update params in request body.

We still need to check object received though since TypeScript does not do exact types. 

https://github.com/microsoft/TypeScript/issues/12936

If interface is: 

```
interface periodUpdateParams {
  name?: string;
  endDate?: string;
}
```

The following object would not generate an error:

```
{
  name: "new name",
  paramWeDontWantUserToChange: "new value"
}
```